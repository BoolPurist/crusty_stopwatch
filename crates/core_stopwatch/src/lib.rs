pub mod now_provider;

mod chrono_duration_serde;
mod stopwatch;
mod timer;

pub use now_provider::{ChronoNow, NowProvider};
pub use stopwatch::Stopwatch;
pub use timer::Timer;

use derive_new::new;
use serde::{Deserialize, Serialize};

#[derive(new, Serialize, Deserialize)]
pub struct LoadedTimers {
    stopwatches: Vec<Stopwatch<ChronoNow>>,
}

#[cfg(test)]
mod testing {
    use chrono::{Duration, TimeZone, Utc};
    use ron::ser::PrettyConfig;

    use super::*;

    #[test]
    fn serilize_stopwatches() {
        let start = Utc.with_ymd_and_hms(2023, 2, 1, 10, 10, 10).unwrap();
        let after_one_day = start + Duration::days(1);
        let input = LoadedTimers::new(vec![
            Stopwatch::from_on(None, start),
            Stopwatch::from_on(None, after_one_day),
        ]);
        let output = ron::ser::to_string_pretty(&input, PrettyConfig::default()).unwrap();

        insta::assert_display_snapshot!(output);
    }
}
