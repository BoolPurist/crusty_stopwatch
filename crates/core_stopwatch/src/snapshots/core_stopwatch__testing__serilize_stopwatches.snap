---
source: crates/core_stopwatch/src/lib.rs
expression: output
---
(
    stopwatches: [
        (
            title: None,
            started: "2023-02-01T10:10:10Z",
            elasped_since_last_pause: 0,
            is_paused: true,
            provider_for_now: (),
        ),
        (
            title: None,
            started: "2023-02-02T10:10:10Z",
            elasped_since_last_pause: 0,
            is_paused: true,
            provider_for_now: (),
        ),
    ],
)
